website/static/js/logTextParser.js

```diff
@@ -0,0 +1,239 @@
+                        return m.component(LogPieces[logComponentName], logObject);
+                    }
+                    return piece;
+                })
+            ]);
+        } else {
+            var message = 'There is no text entry in dictionary for the action :' + logObject.attributes.action;
+            Raven.captureMessage(message, {logObject: logObject});
+        }
+    }
+};
+
+var LogPieces = {
+    // User that took the action
+    user: {
+        view: function (ctrl, logObject) {
+            var userObject = logObject.embeds.user;
+            if(paramIsReturned(userObject, logObject)) {
+                return m('a', {href: userObject.data.links.html}, userObject.data.attributes.full_name);
+            } else {
+                return m('span', 'a user');
+            }
+        }
+    },
+    // Node involved
+    node: {
+        view: function (ctrl, logObject) {
+            var nodeObject = logObject.embeds.nodes;
+            if(paramIsReturned(nodeObject, logObject)){
+                return m('a', {href: nodeObject.data[0].links.html}, nodeObject.data[0].attributes.title);
+            } else {
+                return m('span', 'a project');
+            }
+        }
+    },
+    // Contrubutor list of added, updated etc.
+    contributors: {
+        view: function (ctrl, logObject) {
+            var contributors = logObject.embeds.contributors;
+            if(paramIsReturned(contributors, logObject)) {
+                return contributors.map(function(item){
+                    return m('a', {href: '#'}, 'Person');
+                });
+            }
+            return m('span', 'some users');
+        }
+    },
+    // The tag added to item involved
+    tag: {
+        view: function (ctrl, logObject) {
+            var tag = logObject.attributes.params.tag;
+            if(paramIsReturned(tag, logObject)) {
+               return m('a', {href: '/search/?q=%22' + tag + '%22'}, tag);
+            }
+            return m('span', 'a tag');
+       }
+    },
+    // Node that is linked to the node involved
+    pointer: {
+        view: function (ctrl, logObject) {
+            var linked_node = logObject.embeds.linked_node;
+            if(paramIsReturned(linked_node, logObject)){
+                return m('a', {href: linked_node.data.links.html}, linked_node.data.attributes.title);
+            }
+            return m('span','a project' );
+        }
+    },
+    // Node that acted as template to create a new node involved
+    template: {
+        view: function (ctrl, logObject) {
+            var template_node = logObject.embeds.template_node;
+            if(paramIsReturned(template_node, logObject)){
+                return m('a', {href: template_node.data.links.html}, template_node.data.attributes.title);
+            }
+            return m('span','a project' );
+        }
+    },
+    // The original title of node involved
+    title_original: {
```

Comment:
I think this is trying to match the key names in the returned field from the api for what they are supposed to be. Otherwise we would need to have an object that maps these naming differences. Does this make sense?